### PR TITLE
configure-private-chart-operator

### PR DESCRIPTION
**Pull request by calvix**
adjust cluster-apps-operator to  add support for private clusters and use values to set `chart-operator` to run in private cluster mode

### CODE REVIEW

File: service/controller/resource/clusterconfigmap/types.go [hunk 1]
```diff
@@ -25,6 +25,7 @@ type ClusterValuesConfig struct {
 	ClusterDNSIP  string              `json:"clusterDNSIP"`
 	ClusterID     string              `json:"clusterID"`
 	ClusterCIDR   string              `json:"clusterCIDR"`
+	ExternalDNSIP *string             `json:"externalDNSIP,omitempty"`
```
Comment (calvix) [inline service/controller/resource/clusterconfigmap/types.go:28]: using pointer so we can still  leverage `omitEmpty` but also be able to specify empty string which will override default chart value in `chart-operator` helm chart

**Review by fiunchinho — APPROVED** [review]
Small suggestion, LGTM otherwise.

**Review by fiunchinho — CHANGES_REQUESTED** [review]
Wait no, it's missing a changelog update.

**Comment by calvix** [comment]
changelog fixed

**Comment by calvix** [comment]
@fiunchinho happy with changelog?

**Review by fiunchinho — APPROVED** [review]

**Pull request merged**